app/models/message.rb

```diff
@@ -233,6 +234,20 @@ def ensure_processed_message_content
     self.processed_message_content = message_content&.truncate(150_000)
   end
 
+  # fetch the in_reply_to message and set the external id
+  def ensure_in_reply_to
+    in_reply_to = content_attributes[:in_reply_to]
+    in_reply_to_external_id = content_attributes[:in_reply_to_external_id]
+
+    if in_reply_to.present? && in_reply_to_external_id.blank?
+      message = conversation.messages.find_by(id: in_reply_to)
+      content_attributes[:in_reply_to_external_id] = message.try(:source_id)
+    elsif in_reply_to_external_id.present? && in_reply_to.blank?
+      message = conversation.messages.find_by(source_id: in_reply_to_external_id)
+      content_attributes[:in_reply_to] = message.try(:id)
+    end
+  end
```

Comment:
This is what I was suggesting in the previous PR. This code is repeated everywhere. It could be rewritten as the following. 

```rb
class Messages::InReplyToMessageBuilder
  pattr_initialize [:message!, :in_reply_to!, :in_reply_to_external_id!]

  delegate :conversation, to: :message

  def perform
    if @in_reply_to.present? && @in_reply_to_external_id.blank?
      message.content_attributes[:in_reply_to_external_id] = in_reply_to_message.try(:source_id)
    elsif @in_reply_to_external_id.present? && @in_reply_to.blank?
     message.content_attributes[:in_reply_to] = in_reply_to_message.try(:id)
    end
    message.save!
  end

  private

  def in_reply_to_message
    return @conversation.messages.find_by(id: @in_reply_to) if @in_reply_to.present?

    return @conversation.messages.find_by(source_id: @in_reply_to_external_id) if @in_reply_to_external_id

    nil
  end
end

# ....

def ensure_in_reply_to
  Messages::InReplyToMessageBuilder.new(message: message, in_reply_to: in_reply_to, in_reply_to_external_id:  in_reply_to_external_id).perform
end

```

Reply:
It can be simplified as follows. @scmmishra Make sure you reset both the attributes if in_reply_to message is not present.

```rb
class Messages::InReplyToMessageBuilder
  pattr_initialize [:message!, :in_reply_to!, :in_reply_to_external_id!]

  delegate :conversation, to: :message

  def perform
    set_in_reply_to_attribute if @in_reply_to.present? || @in_reply_to_external_id.present?
  end

  private

  def set_in_reply_to_attribute
    @message.content_attributes[:in_reply_to_external_id] = in_reply_to_message.try(:source_id)
    @message.content_attributes[:in_reply_to] = in_reply_to_message.try(:id)
    @message.save!
  end

  def in_reply_to_message
    return @conversation.messages.find_by(id: @in_reply_to) if @in_reply_to.present?

    return @conversation.messages.find_by(source_id: @in_reply_to_external_id) if @in_reply_to_external_id

    nil
  end
end
```

Reply:
@pranavrajs this is done